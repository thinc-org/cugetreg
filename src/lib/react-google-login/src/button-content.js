import React from 'react'

const ButtonContent = ({ children, icon }) => (
  <span style={{ paddingRight: 10, fontWeight: 500, paddingLeft: icon ? 0 : 10, paddingTop: 10, paddingBottom: 10 }}>
    {children}
  </span>
)

export default ButtonContent
