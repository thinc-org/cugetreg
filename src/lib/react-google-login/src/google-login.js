import React, { useState } from 'react'
import PropTypes from 'prop-types'
import useGoogleLogin from './use-google-login'
import ButtonContent from './button-content'
import { Icon } from './icon'

const GoogleLogin = (props) => {
  const [hovered, setHovered] = useState(false)
  const [active, setActive] = useState(false)
  const {
    onSuccess,
    onAutoLoadFinished,
    onRequest,
    onFailure,
    onScriptLoadFailure,
    tag,
    type,
    className,
    disabledStyle,
    buttonText,
    children,
    render,
    theme,
    icon,
    disabled: disabledProp,
    clientId,
    cookiePolicy,
    loginHint,
    hostedDomain,
    autoLoad,
    isSignedIn,
    fetchBasicProfile,
    redirectUri,
    discoveryDocs,
    uxMode,
    scope,
    accessType,
    responseType,
    jsSrc,
    prompt,
  } = props

  const { signIn, loaded } = useGoogleLogin({
    onSuccess,
    onAutoLoadFinished,
    onRequest,
    onFailure,
    onScriptLoadFailure,
    clientId,
    cookiePolicy,
    loginHint,
    hostedDomain,
    autoLoad,
    isSignedIn,
    fetchBasicProfile,
    redirectUri,
    discoveryDocs,
    uxMode,
    scope,
    accessType,
    responseType,
    jsSrc,
    prompt,
  })
  const disabled = disabledProp || !loaded

  if (render) {
    return render({ onClick: signIn, disabled })
  }

  const initialStyle = {
    backgroundColor: theme === 'dark' ? 'rgb(66, 133, 244)' : '#fff',
    display: 'inline-flex',
    alignItems: 'center',
    color: theme === 'dark' ? '#fff' : 'rgba(0, 0, 0, .54)',
    boxShadow: '0 2px 2px 0 rgba(0, 0, 0, .24), 0 0 1px 0 rgba(0, 0, 0, .24)',
    padding: 0,
    borderRadius: 2,
    border: '1px solid transparent',
    fontSize: 14,
    fontWeight: '500',
    fontFamily: 'Roboto, sans-serif',
  }

  const hoveredStyle = {
    cursor: 'pointer',
    opacity: 0.9,
  }

  const activeStyle = {
    cursor: 'pointer',
    backgroundColor: theme === 'dark' ? '#3367D6' : '#eee',
    color: theme === 'dark' ? '#fff' : 'rgba(0, 0, 0, .54)',
    opacity: 1,
  }

  const defaultStyle = (() => {
    if (disabled) {
      return Object.assign({}, initialStyle, disabledStyle)
    }

    if (active) {
      if (theme === 'dark') {
        return Object.assign({}, initialStyle, activeStyle)
      }

      return Object.assign({}, initialStyle, activeStyle)
    }

    if (hovered) {
      return Object.assign({}, initialStyle, hoveredStyle)
    }

    return initialStyle
  })()
  const googleLoginButton = React.createElement(
    tag,
    {
      onMouseEnter: () => setHovered(true),
      onMouseLeave: () => {
        setHovered(false)
        setActive(false)
      },
      onMouseDown: () => setActive(true),
      onMouseUp: () => setActive(false),
      onClick: signIn,
      style: defaultStyle,
      type,
      disabled,
      className,
    },
    [
      icon && <Icon key={1} active={active} />,
      <ButtonContent icon={icon} key={2}>
        {children || buttonText}
      </ButtonContent>,
    ]
  )

  return googleLoginButton
}

GoogleLogin.propTypes = {
  onSuccess: PropTypes.func.isRequired,
  onFailure: PropTypes.func.isRequired,
  onScriptLoadFailure: PropTypes.func,
  clientId: PropTypes.string.isRequired,
  jsSrc: PropTypes.string,
  onRequest: PropTypes.func,
  buttonText: PropTypes.node,
  scope: PropTypes.string,
  className: PropTypes.string,
  redirectUri: PropTypes.string,
  cookiePolicy: PropTypes.string,
  loginHint: PropTypes.string,
  hostedDomain: PropTypes.string,
  children: PropTypes.node,
  disabledStyle: PropTypes.object,
  fetchBasicProfile: PropTypes.bool,
  prompt: PropTypes.string,
  tag: PropTypes.string,
  autoLoad: PropTypes.bool,
  disabled: PropTypes.bool,
  discoveryDocs: PropTypes.array,
  uxMode: PropTypes.string,
  isSignedIn: PropTypes.bool,
  responseType: PropTypes.string,
  type: PropTypes.string,
  accessType: PropTypes.string,
  render: PropTypes.func,
  theme: PropTypes.string,
  icon: PropTypes.bool,
}

GoogleLogin.defaultProps = {
  type: 'button',
  tag: 'button',
  buttonText: 'Sign in with Google',
  scope: 'profile email',
  accessType: 'online',
  prompt: '',
  cookiePolicy: 'single_host_origin',
  fetchBasicProfile: true,
  isSignedIn: false,
  uxMode: 'popup',
  disabledStyle: {
    opacity: 0.6,
  },
  icon: true,
  theme: 'light',
  onRequest: () => {},
}

export default GoogleLogin
