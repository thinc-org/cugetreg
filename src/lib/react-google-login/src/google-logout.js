import React, { useState } from 'react'
import PropTypes from 'prop-types'
import useGoogleLogout from './use-google-logout'
import ButtonContent from './button-content'
import { Icon } from './icon'

const GoogleLogout = (props) => {
  const [hovered, setHovered] = useState(false)
  const [active, setActive] = useState(false)
  const {
    tag,
    type,
    className,
    disabledStyle,
    buttonText,
    children,
    render,
    theme,
    icon,
    disabled: disabledProp,
    onLogoutSuccess,
    clientId,
    cookiePolicy,
    loginHint,
    hostedDomain,
    fetchBasicProfile,
    redirectUri,
    discoveryDocs,
    onFailure,
    onScriptLoadFailure,
    uxMode,
    scope,
    accessType,
    jsSrc,
  } = props

  const { signOut, loaded } = useGoogleLogout({
    jsSrc,
    onFailure,
    onScriptLoadFailure,
    clientId,
    cookiePolicy,
    loginHint,
    hostedDomain,
    fetchBasicProfile,
    discoveryDocs,
    uxMode,
    redirectUri,
    scope,
    accessType,
    onLogoutSuccess,
  })
  const disabled = disabledProp || !loaded

  if (render) {
    return render({ onClick: signOut, disabled })
  }

  const initialStyle = {
    backgroundColor: theme === 'dark' ? 'rgb(66, 133, 244)' : '#fff',
    display: 'inline-flex',
    alignItems: 'center',
    color: theme === 'dark' ? '#fff' : 'rgba(0, 0, 0, .54)',
    boxShadow: '0 2px 2px 0 rgba(0, 0, 0, .24), 0 0 1px 0 rgba(0, 0, 0, .24)',
    padding: 0,
    borderRadius: 2,
    border: '1px solid transparent',
    fontSize: 14,
    fontWeight: '500',
    fontFamily: 'Roboto, sans-serif',
  }

  const hoveredStyle = {
    cursor: 'pointer',
    opacity: 0.9,
  }

  const activeStyle = {
    cursor: 'pointer',
    backgroundColor: theme === 'dark' ? '#3367D6' : '#eee',
    color: theme === 'dark' ? '#fff' : 'rgba(0, 0, 0, .54)',
    opacity: 1,
  }

  const defaultStyle = (() => {
    if (disabled) {
      return Object.assign({}, initialStyle, disabledStyle)
    }

    if (active) {
      if (theme === 'dark') {
        return Object.assign({}, initialStyle, activeStyle)
      }

      return Object.assign({}, initialStyle, activeStyle)
    }

    if (hovered) {
      return Object.assign({}, initialStyle, hoveredStyle)
    }

    return initialStyle
  })()
  const GoogleLogoutButton = React.createElement(
    tag,
    {
      onMouseEnter: () => setHovered(true),
      onMouseLeave: () => {
        setHovered(false)
        setActive(false)
      },
      onMouseDown: () => setActive(true),
      onMouseUp: () => setActive(false),
      onClick: signOut,
      style: defaultStyle,
      type,
      disabled,
      className,
    },
    [
      icon && <Icon key={1} active={active} />,
      <ButtonContent icon={icon} key={2}>
        {children || buttonText}
      </ButtonContent>,
    ]
  )

  return GoogleLogoutButton
}

GoogleLogout.propTypes = {
  jsSrc: PropTypes.string,
  buttonText: PropTypes.node,
  className: PropTypes.string,
  children: PropTypes.node,
  disabledStyle: PropTypes.object,
  tag: PropTypes.string,
  disabled: PropTypes.bool,
  onLogoutSuccess: PropTypes.func,
  type: PropTypes.string,
  render: PropTypes.func,
  theme: PropTypes.string,
  icon: PropTypes.bool,
  onFailure: PropTypes.func,
  onScriptLoadFailure: PropTypes.func,
}

GoogleLogout.defaultProps = {
  type: 'button',
  tag: 'button',
  buttonText: 'Logout of Google',
  disabledStyle: {
    opacity: 0.6,
  },
  icon: true,
  theme: 'light',
  jsSrc: 'https://apis.google.com/js/api.js',
}

export default GoogleLogout
