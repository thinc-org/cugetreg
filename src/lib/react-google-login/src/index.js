export { default } from './google-login'
export { default as GoogleLogin } from './google-login'
export { default as GoogleLogout } from './google-logout'
export { default as useGoogleLogin } from './use-google-login'
export { default as useGoogleLogout } from './use-google-logout'
