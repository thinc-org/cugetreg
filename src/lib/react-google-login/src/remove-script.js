export const removeScript = (d, id) => {
  const element = d.getElementById(id)

  if (element) {
    element.parentNode.removeChild(element)
  }
}
