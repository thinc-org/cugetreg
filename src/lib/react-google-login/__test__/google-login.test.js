import React from 'react'
import { create } from 'react-test-renderer'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import { GoogleLogin } from '../src'

configure({ adapter: new Adapter() })

describe('Google Login', () => {
  describe('With default props', () => {
    const props = {
      onSuccess() {},
      onFailure() {},
      clientId: '658977310896-knrl3gka66fldh83dao2rhgbblmd4un9.apps.googleusercontent.com',
    }

    test('render the button', () => {
      const component = create(<GoogleLogin {...props} />)
      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    })

    const button = shallow(<GoogleLogin {...props} />)

    test('does not have a class attr', () => {
      expect(button.prop('className')).toEqual(undefined)
    })

    test('has inline styles', () => {
      expect(button.prop('style')).toMatchSnapshot()
    })

    test('displays a button element when tag prop is not set', () => {
      expect(button.type()).toEqual('button')
    })
  })
  describe('With custom text and default props', () => {
    const buttonText = 'buttonText'

    const props = {
      onSuccess() {},
      onFailure() {},
      clientId: '658977310896-knrl3gka66fldh83dao2rhgbblmd4un9.apps.googleusercontent.com',
      buttonText,
    }

    test('render the button', () => {
      const component = create(<GoogleLogin {...props} />)
      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    })

    const button = shallow(<GoogleLogin {...props} />)

    test('does not have a class attr', () => {
      expect(button.prop('className')).toEqual(undefined)
    })

    test('has inline styles', () => {
      expect(button.prop('style')).toMatchSnapshot()
    })

    test('displays a button element when tag prop is not set', () => {
      expect(button.type()).toEqual('button')
    })
  })
  describe('With custom class and default props', () => {
    const className = 'test-class'

    const props = {
      onSuccess() {},
      onFailure() {},
      clientId: '658977310896-knrl3gka66fldh83dao2rhgbblmd4un9.apps.googleusercontent.com',
      className,
    }

    test('render the button', () => {
      const component = create(<GoogleLogin {...props} />)
      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    })

    const button = shallow(<GoogleLogin {...props} />)

    test('does not have a class attr', () => {
      expect(button.prop('className')).toEqual(className)
    })

    test('has inline styles', () => {
      expect(button.prop('style')).toMatchSnapshot()
    })

    test('displays a button element when tag prop is not set', () => {
      expect(button.type()).toEqual('button')
    })
  })
  describe('With custom class and custom style', () => {
    const className = 'test-class'
    const style = { color: 'red' }
    const props = {
      onSuccess() {},
      onFailure() {},
      clientId: '658977310896-knrl3gka66fldh83dao2rhgbblmd4un9.apps.googleusercontent.com',
      className,
      style,
    }

    test('render the button', () => {
      const component = create(<GoogleLogin {...props} />)
      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    })

    const button = shallow(<GoogleLogin {...props} />)

    test('does not have a class attr', () => {
      expect(button.prop('className')).toEqual(className)
    })

    test('has inline styles', () => {
      expect(button.prop('style')).toMatchSnapshot()
    })

    test('displays a button element when tag prop is not set', () => {
      expect(button.type()).toEqual('button')
    })
  })
  describe('With children, custom text, and default props', () => {
    const children = 'test'
    const buttonText = 'buttonText'
    const props = {
      onSuccess() {},
      onFailure() {},
      clientId: '658977310896-knrl3gka66fldh83dao2rhgbblmd4un9.apps.googleusercontent.com',
      buttonText,
    }

    test('render the button', () => {
      const component = create(<GoogleLogin {...props}>{children}</GoogleLogin>)
      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    })

    const button = shallow(<GoogleLogin {...props}>{children}</GoogleLogin>)

    test('does not have a class attr', () => {
      expect(button.prop('className')).toEqual(undefined)
    })

    test('has inline styles', () => {
      expect(button.prop('style')).toMatchSnapshot()
    })

    test('displays a button element when tag prop is not set', () => {
      expect(button.type()).toEqual('button')
    })
  })
  describe('With handles custom tag prop', () => {
    const tag = 'div'

    const props = {
      onSuccess() {},
      onFailure() {},
      clientId: '658977310896-knrl3gka66fldh83dao2rhgbblmd4un9.apps.googleusercontent.com',
      tag,
    }

    test('render the button', () => {
      const component = create(<GoogleLogin {...props} />)
      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    })

    const button = shallow(<GoogleLogin {...props} />)

    test('does not have a class attr', () => {
      expect(button.prop('className')).toEqual(undefined)
    })

    test('has inline styles', () => {
      expect(button.prop('style')).toMatchSnapshot()
    })

    test('displays a button element when tag prop is not set', () => {
      expect(button.type()).toEqual(tag)
    })
  })
})
