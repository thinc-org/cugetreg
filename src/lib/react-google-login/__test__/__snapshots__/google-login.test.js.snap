// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Google Login With children, custom text, and default props has inline styles 1`] = `
Object {
  "alignItems": "center",
  "backgroundColor": "#fff",
  "border": "1px solid transparent",
  "borderRadius": 2,
  "boxShadow": "0 2px 2px 0 rgba(0, 0, 0, .24), 0 0 1px 0 rgba(0, 0, 0, .24)",
  "color": "rgba(0, 0, 0, .54)",
  "display": "inline-flex",
  "fontFamily": "Roboto, sans-serif",
  "fontSize": 14,
  "fontWeight": "500",
  "opacity": 0.6,
  "padding": 0,
}
`;

exports[`Google Login With children, custom text, and default props render the button 1`] = `
<button
  disabled={true}
  onClick={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "border": "1px solid transparent",
      "borderRadius": 2,
      "boxShadow": "0 2px 2px 0 rgba(0, 0, 0, .24), 0 0 1px 0 rgba(0, 0, 0, .24)",
      "color": "rgba(0, 0, 0, .54)",
      "display": "inline-flex",
      "fontFamily": "Roboto, sans-serif",
      "fontSize": 14,
      "fontWeight": "500",
      "opacity": 0.6,
      "padding": 0,
    }
  }
  type="button"
>
  <div
    style={
      Object {
        "background": "#fff",
        "borderRadius": 2,
        "marginRight": 10,
        "padding": 10,
      }
    }
  >
    <svg
      height="18"
      width="18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="#000"
        fillRule="evenodd"
      >
        <path
          d="M9 3.48c1.69 0 2.83.73 3.48 1.34l2.54-2.48C13.46.89 11.43 0 9 0 5.48 0 2.44 2.02.96 4.96l2.91 2.26C4.6 5.05 6.62 3.48 9 3.48z"
          fill="#EA4335"
        />
        <path
          d="M17.64 9.2c0-.74-.06-1.28-.19-1.84H9v3.34h4.96c-.1.83-.64 2.08-1.84 2.92l2.84 2.2c1.7-1.57 2.68-3.88 2.68-6.62z"
          fill="#4285F4"
        />
        <path
          d="M3.88 10.78A5.54 5.54 0 0 1 3.58 9c0-.62.11-1.22.29-1.78L.96 4.96A9.008 9.008 0 0 0 0 9c0 1.45.35 2.82.96 4.04l2.92-2.26z"
          fill="#FBBC05"
        />
        <path
          d="M9 18c2.43 0 4.47-.8 5.96-2.18l-2.84-2.2c-.76.53-1.78.9-3.12.9-2.38 0-4.4-1.57-5.12-3.74L.97 13.04C2.45 15.98 5.48 18 9 18z"
          fill="#34A853"
        />
        <path
          d="M0 0h18v18H0z"
          fill="none"
        />
      </g>
    </svg>
  </div>
  <span
    style={
      Object {
        "fontWeight": 500,
        "paddingBottom": 10,
        "paddingLeft": 0,
        "paddingRight": 10,
        "paddingTop": 10,
      }
    }
  >
    test
  </span>
</button>
`;

exports[`Google Login With custom class and custom style has inline styles 1`] = `
Object {
  "alignItems": "center",
  "backgroundColor": "#fff",
  "border": "1px solid transparent",
  "borderRadius": 2,
  "boxShadow": "0 2px 2px 0 rgba(0, 0, 0, .24), 0 0 1px 0 rgba(0, 0, 0, .24)",
  "color": "rgba(0, 0, 0, .54)",
  "display": "inline-flex",
  "fontFamily": "Roboto, sans-serif",
  "fontSize": 14,
  "fontWeight": "500",
  "opacity": 0.6,
  "padding": 0,
}
`;

exports[`Google Login With custom class and custom style render the button 1`] = `
<button
  className="test-class"
  disabled={true}
  onClick={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "border": "1px solid transparent",
      "borderRadius": 2,
      "boxShadow": "0 2px 2px 0 rgba(0, 0, 0, .24), 0 0 1px 0 rgba(0, 0, 0, .24)",
      "color": "rgba(0, 0, 0, .54)",
      "display": "inline-flex",
      "fontFamily": "Roboto, sans-serif",
      "fontSize": 14,
      "fontWeight": "500",
      "opacity": 0.6,
      "padding": 0,
    }
  }
  type="button"
>
  <div
    style={
      Object {
        "background": "#fff",
        "borderRadius": 2,
        "marginRight": 10,
        "padding": 10,
      }
    }
  >
    <svg
      height="18"
      width="18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="#000"
        fillRule="evenodd"
      >
        <path
          d="M9 3.48c1.69 0 2.83.73 3.48 1.34l2.54-2.48C13.46.89 11.43 0 9 0 5.48 0 2.44 2.02.96 4.96l2.91 2.26C4.6 5.05 6.62 3.48 9 3.48z"
          fill="#EA4335"
        />
        <path
          d="M17.64 9.2c0-.74-.06-1.28-.19-1.84H9v3.34h4.96c-.1.83-.64 2.08-1.84 2.92l2.84 2.2c1.7-1.57 2.68-3.88 2.68-6.62z"
          fill="#4285F4"
        />
        <path
          d="M3.88 10.78A5.54 5.54 0 0 1 3.58 9c0-.62.11-1.22.29-1.78L.96 4.96A9.008 9.008 0 0 0 0 9c0 1.45.35 2.82.96 4.04l2.92-2.26z"
          fill="#FBBC05"
        />
        <path
          d="M9 18c2.43 0 4.47-.8 5.96-2.18l-2.84-2.2c-.76.53-1.78.9-3.12.9-2.38 0-4.4-1.57-5.12-3.74L.97 13.04C2.45 15.98 5.48 18 9 18z"
          fill="#34A853"
        />
        <path
          d="M0 0h18v18H0z"
          fill="none"
        />
      </g>
    </svg>
  </div>
  <span
    style={
      Object {
        "fontWeight": 500,
        "paddingBottom": 10,
        "paddingLeft": 0,
        "paddingRight": 10,
        "paddingTop": 10,
      }
    }
  >
    Sign in with Google
  </span>
</button>
`;

exports[`Google Login With custom class and default props has inline styles 1`] = `
Object {
  "alignItems": "center",
  "backgroundColor": "#fff",
  "border": "1px solid transparent",
  "borderRadius": 2,
  "boxShadow": "0 2px 2px 0 rgba(0, 0, 0, .24), 0 0 1px 0 rgba(0, 0, 0, .24)",
  "color": "rgba(0, 0, 0, .54)",
  "display": "inline-flex",
  "fontFamily": "Roboto, sans-serif",
  "fontSize": 14,
  "fontWeight": "500",
  "opacity": 0.6,
  "padding": 0,
}
`;

exports[`Google Login With custom class and default props render the button 1`] = `
<button
  className="test-class"
  disabled={true}
  onClick={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "border": "1px solid transparent",
      "borderRadius": 2,
      "boxShadow": "0 2px 2px 0 rgba(0, 0, 0, .24), 0 0 1px 0 rgba(0, 0, 0, .24)",
      "color": "rgba(0, 0, 0, .54)",
      "display": "inline-flex",
      "fontFamily": "Roboto, sans-serif",
      "fontSize": 14,
      "fontWeight": "500",
      "opacity": 0.6,
      "padding": 0,
    }
  }
  type="button"
>
  <div
    style={
      Object {
        "background": "#fff",
        "borderRadius": 2,
        "marginRight": 10,
        "padding": 10,
      }
    }
  >
    <svg
      height="18"
      width="18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="#000"
        fillRule="evenodd"
      >
        <path
          d="M9 3.48c1.69 0 2.83.73 3.48 1.34l2.54-2.48C13.46.89 11.43 0 9 0 5.48 0 2.44 2.02.96 4.96l2.91 2.26C4.6 5.05 6.62 3.48 9 3.48z"
          fill="#EA4335"
        />
        <path
          d="M17.64 9.2c0-.74-.06-1.28-.19-1.84H9v3.34h4.96c-.1.83-.64 2.08-1.84 2.92l2.84 2.2c1.7-1.57 2.68-3.88 2.68-6.62z"
          fill="#4285F4"
        />
        <path
          d="M3.88 10.78A5.54 5.54 0 0 1 3.58 9c0-.62.11-1.22.29-1.78L.96 4.96A9.008 9.008 0 0 0 0 9c0 1.45.35 2.82.96 4.04l2.92-2.26z"
          fill="#FBBC05"
        />
        <path
          d="M9 18c2.43 0 4.47-.8 5.96-2.18l-2.84-2.2c-.76.53-1.78.9-3.12.9-2.38 0-4.4-1.57-5.12-3.74L.97 13.04C2.45 15.98 5.48 18 9 18z"
          fill="#34A853"
        />
        <path
          d="M0 0h18v18H0z"
          fill="none"
        />
      </g>
    </svg>
  </div>
  <span
    style={
      Object {
        "fontWeight": 500,
        "paddingBottom": 10,
        "paddingLeft": 0,
        "paddingRight": 10,
        "paddingTop": 10,
      }
    }
  >
    Sign in with Google
  </span>
</button>
`;

exports[`Google Login With custom text and default props has inline styles 1`] = `
Object {
  "alignItems": "center",
  "backgroundColor": "#fff",
  "border": "1px solid transparent",
  "borderRadius": 2,
  "boxShadow": "0 2px 2px 0 rgba(0, 0, 0, .24), 0 0 1px 0 rgba(0, 0, 0, .24)",
  "color": "rgba(0, 0, 0, .54)",
  "display": "inline-flex",
  "fontFamily": "Roboto, sans-serif",
  "fontSize": 14,
  "fontWeight": "500",
  "opacity": 0.6,
  "padding": 0,
}
`;

exports[`Google Login With custom text and default props render the button 1`] = `
<button
  disabled={true}
  onClick={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "border": "1px solid transparent",
      "borderRadius": 2,
      "boxShadow": "0 2px 2px 0 rgba(0, 0, 0, .24), 0 0 1px 0 rgba(0, 0, 0, .24)",
      "color": "rgba(0, 0, 0, .54)",
      "display": "inline-flex",
      "fontFamily": "Roboto, sans-serif",
      "fontSize": 14,
      "fontWeight": "500",
      "opacity": 0.6,
      "padding": 0,
    }
  }
  type="button"
>
  <div
    style={
      Object {
        "background": "#fff",
        "borderRadius": 2,
        "marginRight": 10,
        "padding": 10,
      }
    }
  >
    <svg
      height="18"
      width="18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="#000"
        fillRule="evenodd"
      >
        <path
          d="M9 3.48c1.69 0 2.83.73 3.48 1.34l2.54-2.48C13.46.89 11.43 0 9 0 5.48 0 2.44 2.02.96 4.96l2.91 2.26C4.6 5.05 6.62 3.48 9 3.48z"
          fill="#EA4335"
        />
        <path
          d="M17.64 9.2c0-.74-.06-1.28-.19-1.84H9v3.34h4.96c-.1.83-.64 2.08-1.84 2.92l2.84 2.2c1.7-1.57 2.68-3.88 2.68-6.62z"
          fill="#4285F4"
        />
        <path
          d="M3.88 10.78A5.54 5.54 0 0 1 3.58 9c0-.62.11-1.22.29-1.78L.96 4.96A9.008 9.008 0 0 0 0 9c0 1.45.35 2.82.96 4.04l2.92-2.26z"
          fill="#FBBC05"
        />
        <path
          d="M9 18c2.43 0 4.47-.8 5.96-2.18l-2.84-2.2c-.76.53-1.78.9-3.12.9-2.38 0-4.4-1.57-5.12-3.74L.97 13.04C2.45 15.98 5.48 18 9 18z"
          fill="#34A853"
        />
        <path
          d="M0 0h18v18H0z"
          fill="none"
        />
      </g>
    </svg>
  </div>
  <span
    style={
      Object {
        "fontWeight": 500,
        "paddingBottom": 10,
        "paddingLeft": 0,
        "paddingRight": 10,
        "paddingTop": 10,
      }
    }
  >
    buttonText
  </span>
</button>
`;

exports[`Google Login With default props has inline styles 1`] = `
Object {
  "alignItems": "center",
  "backgroundColor": "#fff",
  "border": "1px solid transparent",
  "borderRadius": 2,
  "boxShadow": "0 2px 2px 0 rgba(0, 0, 0, .24), 0 0 1px 0 rgba(0, 0, 0, .24)",
  "color": "rgba(0, 0, 0, .54)",
  "display": "inline-flex",
  "fontFamily": "Roboto, sans-serif",
  "fontSize": 14,
  "fontWeight": "500",
  "opacity": 0.6,
  "padding": 0,
}
`;

exports[`Google Login With default props render the button 1`] = `
<button
  disabled={true}
  onClick={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "border": "1px solid transparent",
      "borderRadius": 2,
      "boxShadow": "0 2px 2px 0 rgba(0, 0, 0, .24), 0 0 1px 0 rgba(0, 0, 0, .24)",
      "color": "rgba(0, 0, 0, .54)",
      "display": "inline-flex",
      "fontFamily": "Roboto, sans-serif",
      "fontSize": 14,
      "fontWeight": "500",
      "opacity": 0.6,
      "padding": 0,
    }
  }
  type="button"
>
  <div
    style={
      Object {
        "background": "#fff",
        "borderRadius": 2,
        "marginRight": 10,
        "padding": 10,
      }
    }
  >
    <svg
      height="18"
      width="18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="#000"
        fillRule="evenodd"
      >
        <path
          d="M9 3.48c1.69 0 2.83.73 3.48 1.34l2.54-2.48C13.46.89 11.43 0 9 0 5.48 0 2.44 2.02.96 4.96l2.91 2.26C4.6 5.05 6.62 3.48 9 3.48z"
          fill="#EA4335"
        />
        <path
          d="M17.64 9.2c0-.74-.06-1.28-.19-1.84H9v3.34h4.96c-.1.83-.64 2.08-1.84 2.92l2.84 2.2c1.7-1.57 2.68-3.88 2.68-6.62z"
          fill="#4285F4"
        />
        <path
          d="M3.88 10.78A5.54 5.54 0 0 1 3.58 9c0-.62.11-1.22.29-1.78L.96 4.96A9.008 9.008 0 0 0 0 9c0 1.45.35 2.82.96 4.04l2.92-2.26z"
          fill="#FBBC05"
        />
        <path
          d="M9 18c2.43 0 4.47-.8 5.96-2.18l-2.84-2.2c-.76.53-1.78.9-3.12.9-2.38 0-4.4-1.57-5.12-3.74L.97 13.04C2.45 15.98 5.48 18 9 18z"
          fill="#34A853"
        />
        <path
          d="M0 0h18v18H0z"
          fill="none"
        />
      </g>
    </svg>
  </div>
  <span
    style={
      Object {
        "fontWeight": 500,
        "paddingBottom": 10,
        "paddingLeft": 0,
        "paddingRight": 10,
        "paddingTop": 10,
      }
    }
  >
    Sign in with Google
  </span>
</button>
`;

exports[`Google Login With handles custom tag prop has inline styles 1`] = `
Object {
  "alignItems": "center",
  "backgroundColor": "#fff",
  "border": "1px solid transparent",
  "borderRadius": 2,
  "boxShadow": "0 2px 2px 0 rgba(0, 0, 0, .24), 0 0 1px 0 rgba(0, 0, 0, .24)",
  "color": "rgba(0, 0, 0, .54)",
  "display": "inline-flex",
  "fontFamily": "Roboto, sans-serif",
  "fontSize": 14,
  "fontWeight": "500",
  "opacity": 0.6,
  "padding": 0,
}
`;

exports[`Google Login With handles custom tag prop render the button 1`] = `
<div
  disabled={true}
  onClick={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "border": "1px solid transparent",
      "borderRadius": 2,
      "boxShadow": "0 2px 2px 0 rgba(0, 0, 0, .24), 0 0 1px 0 rgba(0, 0, 0, .24)",
      "color": "rgba(0, 0, 0, .54)",
      "display": "inline-flex",
      "fontFamily": "Roboto, sans-serif",
      "fontSize": 14,
      "fontWeight": "500",
      "opacity": 0.6,
      "padding": 0,
    }
  }
  type="button"
>
  <div
    style={
      Object {
        "background": "#fff",
        "borderRadius": 2,
        "marginRight": 10,
        "padding": 10,
      }
    }
  >
    <svg
      height="18"
      width="18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="#000"
        fillRule="evenodd"
      >
        <path
          d="M9 3.48c1.69 0 2.83.73 3.48 1.34l2.54-2.48C13.46.89 11.43 0 9 0 5.48 0 2.44 2.02.96 4.96l2.91 2.26C4.6 5.05 6.62 3.48 9 3.48z"
          fill="#EA4335"
        />
        <path
          d="M17.64 9.2c0-.74-.06-1.28-.19-1.84H9v3.34h4.96c-.1.83-.64 2.08-1.84 2.92l2.84 2.2c1.7-1.57 2.68-3.88 2.68-6.62z"
          fill="#4285F4"
        />
        <path
          d="M3.88 10.78A5.54 5.54 0 0 1 3.58 9c0-.62.11-1.22.29-1.78L.96 4.96A9.008 9.008 0 0 0 0 9c0 1.45.35 2.82.96 4.04l2.92-2.26z"
          fill="#FBBC05"
        />
        <path
          d="M9 18c2.43 0 4.47-.8 5.96-2.18l-2.84-2.2c-.76.53-1.78.9-3.12.9-2.38 0-4.4-1.57-5.12-3.74L.97 13.04C2.45 15.98 5.48 18 9 18z"
          fill="#34A853"
        />
        <path
          d="M0 0h18v18H0z"
          fill="none"
        />
      </g>
    </svg>
  </div>
  <span
    style={
      Object {
        "fontWeight": 500,
        "paddingBottom": 10,
        "paddingLeft": 0,
        "paddingRight": 10,
        "paddingTop": 10,
      }
    }
  >
    Sign in with Google
  </span>
</div>
`;
